### PR TITLE
[READY FOR REVIEW] Login/Register Rhd 970

### PR DESCRIPTION
**Pull request by RubyGemTSL**

### CODE REVIEW

**Comment by RubyGemTSL** [comment]
retest this please?

**Comment by RubyGemTSL** [comment]
retest this please?

**Comment by RubyGemTSL** [comment]
retest this please?

**Comment by ollyjshaw** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
retest this please?

File: _config/site.yml [hunk 1]
```diff
@@ -56,6 +56,8 @@ profiles:
     dcp_base_protocol_relative_url: //dcpbeta-searchisko.rhcloud.com/
     dcp2_base_url: http://dcp2-searchisko.rhcloud.com/
     dcp2_base_protocol_relative_url: //dcp2-searchisko.rhcloud.com/
+    keycloak_account_url: https://it-developers.stage.redhat.com/auth/realms/rhd/account/
+    keycloak_auth_url: https://it-developers.stage.redhat.com/auth/
```
Comment (paulrobinson) [inline _config/site.yml]: You will also need to use the staging download manager, as they keycloak and DL-man need to talk to each other.

Comment (paulrobinson) [inline _config/site.yml]: And that is blocked on the merging: https://github.com/redhat-developer/developers.redhat.com/pull/332, which will hopefully happen this week.

**Comment by paulrobinson** [comment]
@ollyjshaw can you review please.

**Comment by paulrobinson** [comment]
I've had a quick look. Only issue I found is the one I mentioned above.

File: _cucumber/lib/pages/forgot_password_page.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  element :confirmation, '#kc-feedback-text'
+
+
+  def initialize(driver)
+    super
+    loaded?('Forgot Password | Red Hat Developers')
+    wait_for_reset_password_page && wait_for_reset_password_form
+  end
+
+  def enter_username_or_email(email)
```
Comment (ollyjshaw) [inline _cucumber/lib/pages/forgot_password_page.rb]: I think the method name is misleading here.

**Comment by ollyjshaw** [comment]
:+1: from me. Just one minor comment.

File: _config/site.yml (hunk 1 above)
Comment (paulrobinson) [inline _config/site.yml]: You can update the config to use the staging dl-man for all non-production configs.

**Comment by RubyGemTSL** [comment]
@ollyjshaw can you have a quick look over my changes, please? If you're OK with them this can be merged.

**Comment by ollyjshaw** [comment]
:+1:

**Comment by paulrobinson** [comment]
The downloads are still going to production dl-man. This is why: https://github.com/redhat-developer/developers.redhat.com/blob/master/javascripts/downloads.js#L3. You can fix that by paramaterizing the url like this: https://github.com/redhat-developer/developers.redhat.com/blob/master/register.html.slim#L11

**Comment by ollyjshaw** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@paulrobinson the change you suggested (as discussed this morning) breaks the latest download link. Instead of 'Download latest' you see a 'Fetching latest downloads' link.

File: javascripts/downloads.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 app.downloads = {};
 
-app.downloads.url = 'https://developers.redhat.com/download-manager/rest/available/';
+app.downloads.url = '#{site.base_url}/download-manager/rest/available/';
```
Comment (paulrobinson) [inline javascripts/downloads.js]: Try:

      app.downloads.url = app.baseUrl + '/download-manager/rest/available/';

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@paulrobinson your suggestion should fix this, but its still failing - maybe this is related to https://issues.jboss.org/browse/RHD-1372

Is download manager down in staging? For example if I try https://it-developers.stage.redhat.com/download-manager/rest/available/ I get nothing returned.

**Comment by paulrobinson** [comment]
I see the problem:

      The requested URL /pr/657/build/377/download-manager/rest/available/fuse was not found on this server.

Sorry, I wasn't thinking straight. You need to pass in the URL of the download-manager, not the base URL of the site. If there is no variable available in JS, then you can add it to namespace.js, as that file is templated by Awestruct. I can explain this to you in more detail if this doesn't make sense.

**Comment by RubyGemTSL** [comment]
retest this please

**Comment by RubyGemTSL** [comment]
@paulrobinson so I need to revert back to the initial URL?

**Comment by paulrobinson** [comment]
More specifically. 

Add a 'download_manager_base_url' variable underneath every occurrence of the 'download_manager_file_base_url' variable in site.yml. Set it to '//developers.redhat.com/' or '//it-developers.stage.redhat.com' depending on wether staging or production DL-man is used.

Add the following to namespace.js:

      /*
       Download Manager Base URL
       */
      app.downloadManagerBaseUrl = "#{site.download_manager_base_url}";

And then change the URL to use the app.downloadManagerBaseUrl variable.

File: _config/site.yml [hunk 4]
```diff
@@ -106,6 +112,8 @@ profiles:
     dcp_base_url: http://<%= ENV['STAGE_SEARCHISKO_HOST'] %>:<%= ENV['STAGE_SEARCHISKO_PORT'] %>/
     dcp2_base_url: http://dcp2-searchisko.rhcloud.com/
     dcp2_base_protocol_relative_url: //dcp2-searchisko.rhcloud.com/
+    keycloak_account_url: https://it-developers.stage.redhat.com/auth/realms/rhd/account/
+    keycloak_auth_url: https://it-developers.stage.redhat.com/auth/
     metrics: false
```
Comment (paulrobinson) [inline _config/site.yml]: you need to overide the dl-man URLs in here.

**Comment by RubyGemTSL** [comment]
@paulrobinson - All checks have passed!

**Pull request merged**